tests/unit/test_collector.py

```diff
@@ -417,6 +398,66 @@ def test_ensure_quoted_url(url: str, clean_url: str) -> None:
     assert _ensure_quoted_url(url) == clean_url
 
 
+# versions containing fix/backport from https://github.com/python/cpython/pull/113563
+has_new_urlun_behaviour = (
+    # https://github.com/python/cpython/commit/387ff96e95b9f8a8cc7e646523ba3175b1350669
+    (sys.version_info[:2] == (3, 12) and sys.version_info >= (3, 12, 4))
+    or
+    # https://github.com/python/cpython/commit/872000606271c52d989e53fe4cc9904343d81855
+    (sys.version_info[:2] == (3, 13) and sys.version_info >= (3, 13, 0, "beta", 3))
+)
+
+
+@pytest.mark.skipif(
+    sys.platform != "win32" or has_new_urlun_behaviour,
+    reason="testing windows behaviour on older CPython",
+)
+@pytest.mark.parametrize(
+    ("url", "clean_url"),
+    (
+        (
+            # URL with Windows drive letter. The `:` after the drive
+            # letter should not be quoted. The trailing `/` should be
+            # removed.
+            "file:///T:/path/with spaces/",
+            "file:///T:/path/with%20spaces",
+        ),
+        (
+            # Test a VCS URL with a Windows drive letter and revision.
+            "git+file:///T:/with space/repo.git@1.0#egg=my-package-1.0",
+            "git+file:///T:/with%20space/repo.git@1.0#egg=my-package-1.0",
+        ),
+    ),
+)
+def test_ensure_quoted_url_windows_old(url: str, clean_url: str) -> None:
```

Comment:
`*_old` and `*_new` are not really the most informative test names, but also I think they're clear enough/not too significant either way